### ISSUE TITLE
Master compilation fails

### ISSUE THREAD

**Issue by akdjka** [ja2-stracciatella/ja2-stracciatella#758]
Snipplet of build log:
```
--> src/stracciatella.rs:7:1
  |
7 | extern crate serde_derive;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot find derive macro `Deserialize` in this scope
 --> src/config/scaling_quality.rs:8:52
  |
8 | #[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
  |                                                    ^^^^^^^^^^^

error: cannot find derive macro `Serialize` in this scope
 --> src/config/scaling_quality.rs:8:41
  |
8 | #[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
  |                                         ^^^^^^^^^

error: cannot find derive macro `Deserialize` in this scope
 --> src/config/vanilla_version.rs:7:52
  |
7 | #[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
  |                                                    ^^^^^^^^^^^

error: cannot find derive macro `Serialize` in this scope
 --> src/config/vanilla_version.rs:7:41
  |
7 | #[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
  |                                         ^^^^^^^^^

error[E0658]: access to extern crates through prelude is experimental (see issue #44660)
  --> src/config/resolution.rs:47:60
   |
47 |         Resolution::from_str(&deserialized_string).map_err(serde::de::Error::custom)
   |                                                            ^^^^^

error: cannot find derive macro `Deserialize` in this scope
  --> src/config/ja2_json.rs:15:39
   |
15 | #[derive(Debug, PartialEq, Serialize, Deserialize)]
   |                                       ^^^^^^^^^^^

error: cannot find derive macro `Serialize` in this scope
  --> src/config/ja2_json.rs:15:28
   |
15 | #[derive(Debug, PartialEq, Serialize, Deserialize)]
   |                            ^^^^^^^^^

error[E0658]: access to extern crates through prelude is experimental (see issue #44660)
  --> src/config/ja2_json.rs:47:27
   |
47 |             .and_then(|f| serde_json::from_reader(f).map_err(|s| format!("Error parsing ja2.json config file: {}", s)))
   |                           ^^^^^^^^^^

error[E0658]: access to extern crates through prelude is experimental (see issue #44660)
  --> src/config/ja2_json.rs:98:20
   |
98 |         let json = serde_json::to_string_pretty(&content).map_err(|s| format!("Error creating contents of ja2.json config file: {}", s.description()))?;
   |                    ^^^^^^^^^^

error: aborting due to 9 previous errors

For more information about this error, try `rustc --explain E0658`.
error: Could not compile `stracciatella`.
```
Note that 0.16.1 compiles fine.
What else should I provide?

**Comment by lynxlynxlynx** [comment]
I think it's a rust version issue, but @selaux will know for sure, including the details.

**Comment by akdjka** [comment]
I believe if there are special rust requirements they should be documented. Right now the build instructions are not very specific when it comes to compiler choice....

**Comment by lynxlynxlynx** [comment]
We can probably add a minimal version check to cmake.

**Comment by RelayrMA** [comment]
I believe it's actually about the project requiring a nightly now. At least rustc --explain E0658 makes me believe so.
I have rust 1.29.1, not the latest but not old either.

**Comment by RelayrMA** [comment]
I believe it's about the project requiring Nightly now.

**Comment by selaux** [comment]
Hi, we dont require nightly to build, but the current stable version of rust. I checked while current stable is `1.32`, but it also builds with `1.30`.

In order to adress this I created https://github.com/ja2-stracciatella/ja2-stracciatella/pull/759, which pins the rust version when using rustup and makes the required version more clear. It also pins dependencies so we don't get anything unexpected there.
